internal/graph/check.go

```diff
@@ -667,6 +667,25 @@ func (c *LocalChecker) hasCycle(req *ResolveCheckRequest) bool {
 // nolint:unused
 type usersetsMapType map[string]storage.SortedSet
 
+// return whether any of the iterator ID is in sorted set
+func tupleIDInSortedSet(ctx context.Context, filteredIter *storage.ConditionsFilteredTupleKeyIterator, objectIDs storage.SortedSet) (bool, error) {
+	for {
+		t, err := filteredIter.Next(ctx)
+		if err != nil {
+			if errors.Is(err, storage.ErrIteratorDone) {
+				break
+			}
+			return false, err
+		}
+
+		_, objectID := tuple.SplitObject(t.GetObject())
+		if objectIDs.Exists(objectID) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
```

Comment:
**(suggestion):** We could further reduce complexity here and improve on the description. 

```suggestion
// tupleIDInSortedSet checks if any tuple object ID returned by the 
// iterator exists in the provided sorted set of object IDs.
func tupleIDInSortedSet(
	ctx context.Context,
	filteredIter *storage.ConditionsFilteredTupleKeyIterator,
	objectIDs storage.SortedSet,
) (bool, error) {
	for {
		t, err := filteredIter.Next(ctx)
		if errors.Is(err, storage.ErrIteratorDone) {
			return false, nil
		}
		if err != nil {
			return false, err
		}

		_, objectID := tuple.SplitObject(t.GetObject())
		if objectIDs.Exists(objectID) {
			return true, nil
		}
	}
}
```